src/common/mfu_flist_copy.c

```diff
@@ -1847,18 +1864,30 @@ static int mfu_copy_file_normal(
     }
 #endif
 
-    /* if we wrote the last chunk, truncate the file */
+    /* if sparse or using O_DIRECT, we already truncated the file at the beginning, we do not need
+     * to do it again. */
+    if (copy_opts->sparse || !copy_opts->direct)
+	    return 0;
```

Comment:
What if using sparse and O_DIRECT? Should this instead be?
```
    if (!copy_opts->direct)
	    return 0;
```

Reply:
yes indeed.. not sure why i had sparse there